Quick Queue: Add option to place button on the left, Fix for new version

D3SOX opened the pull request:
Resolves #43 
Resolves #45

<img width="221" height="46" alt="image" src="https://github.com/user-attachments/assets/d272efc4-b78c-4005-98e1-7cd2bbb5a55e" />

<img width="1374" height="606" alt="image" src="https://github.com/user-attachments/assets/c7580c40-3866-4667-995a-0b7ea5b0b83c" />

D3SOX committed:
feat(quick-queue): add option to place button on the left

D3SOX committed:
fix(quick-queue): enhance node observation, resilient React props retrieval

D3SOX commented:
Never mind I just pushed a fix for the new version so I think a merger would be good.

ohitstom commented:
> Never mind I just pushed a fix for the new version so I think a merger would be good.

only concerns with this fix is the overuse of classmap class names, I coded it to use very unspecific selectors (or few of them) so that maintenance would become simply changing those selectors

realistically I don't have time to maintain this so any fix would need not constant changing of class names, I'll hold off on merge until someone's tested this in the issue or I get some time to tomorrow

ohitstom commented:
putting this on hold, we really need a seperate PR bringing functionality up to scratch before a feature pr like this - if you are able to do this please by all means.

if not archive this pr and i'll use it and add you as a contributor to the new pr if i get around to it 👍🏻

D3SOX commented:
> bringing functionality up to scratch

Not entirely sure what you mean by that.

I'm currently still running this on the old version. As for the fix I can try again once the new version is available for Linux
Otherwise, if you get around doing it before just close my PR and use it as reference I guess

D3SOX commented:
Working on a fix right now. This PR should be ready soon.

D3SOX committed:
fix: find correct uri in newer spotify versions

Now it also no longer adds the icon if it doesn't find the URI avoiding the confusion like in the issue

D3SOX committed:
refactor: revert unnecessary changes

D3SOX committed:
fix: improve props retrieval error handling

ohitstom commented:
code looks great, can you give it a test on the recommended versions too for me and ensure no weird errors (though your code seems like it should be fine) then i'll go ahead and merge tysm!

https://github.com/ohitstom/spicetify-extensions?tab=readme-ov-file#%EF%B8%8F-recommended-versions

ohitstom commented:
I've updated the main readme, did you want to add your name in credits somewhere (next to mine somewhere) and add a section in the extensions readme copying the main one with the versions you would personally recommend given you are the latest contributor

D3SOX commented:
I can't even find a download for such old Spotify verisons as 1.2.51 anymore.
https://repository.spotify.com/pool/non-free/s/spotify-client/
So I can't verify that sorry

D3SOX committed:
docs(quickQueue):update readme with versions and setting

D3SOX committed:
docs(quickQueue): add back missing header

ohitstom commented:
> Author

https://loadspot.pages.dev/

D3SOX commented:
Thanks, I verified it still works on 1.2.50.335.g5e2860a8 (as there is no 1.2.51 for Linux)

D3SOX committed:
docs(quickQueue): update supported versions

D3SOX committed:
chore: remove trailing space

D3SOX commented:
> did you want to add your name in credits somewhere (next to mine somewhere)

Regarding that I am not sure where you mean "next to mine" but not very important anyways. If you want you can put a little notice somewhere

ohitstom commented:
alrighty give me one last go ahead when ready and i'll merge

D3SOX commented:
Yeah go ahead it's fine

ohitstom committed:
Merge branch 'main' into quick-queue/left-placement
